packages/web/tests/components/DiscordCheck/DiscordButtonCheck.test.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import fetchMock from 'fetch-mock-jest';
+import userEvent from '@testing-library/user-event';
+import { DiscordButtonCheck } from '../../../src/components/DiscordCheck';
+import { act, render, screen } from '../../testUtils/testTools';
+
+jest.mock('next/router', () => ({
+  useRouter() {
+    return {
+      query: '',
+    };
+  },
+}));
+
+const useRouter = jest.spyOn(require('next/router'), 'useRouter');
+
+interface User {
+  discordId?: string;
+}
+
+export interface UserProfileProps {
+  user: User;
+}
+
+const sampleUser: UserProfileProps['user'] = {
+  discordId: undefined,
+};
+
+const sampleUser2: UserProfileProps['user'] = {
+  discordId: '34523452345',
+};
```

Comment:
`DiscordButtonCheck` expects a `User` so, there's no need to use the `UserProfileProps`.

```suggestion
const sampleUser: User  = {
  discordId: undefined,
};

const sampleUser2: User = {
  discordId: '34523452345',
};
```